digitalocean/resource_digitalocean_monitor_alert.go

```diff
@@ -0,0 +1,291 @@
+				Type:         schema.TypeFloat,
+				Required:     true,
+				ValidateFunc: validation.FloatAtLeast(0),
+			},
+
+			"tags": tagsSchema(),
+
+			"alerts": {
+				Type:        schema.TypeList,
+				Computed:    false,
+				Required:    false,
+				Description: "List with details how to notify about the alert. Support for Slack or email.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"slack": {
+							Type:     schema.TypeList,
+							Required: false,
+							Computed: false,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"channel": {
+										Type:         schema.TypeList,
+										Computed:     false,
+										Required:     true,
+										Description:  "The Slack channel to send alerts to",
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"url": {
+										Type:             schema.TypeString,
+										Computed:         false,
+										DiffSuppressFunc: CaseSensitive,
+										Description:      "The webhook URL for Slack",
+										ValidateFunc:     validation.StringIsNotEmpty,
+									},
+								},
+							},
+						},
+						"email": {
+							Type:         schema.TypeList,
+							Computed:     false,
+							Required:     false,
+							ValidateFunc: validation.ListOfUniqueStrings,
+						},
+					},
+				},
+			},
+
+			"entities": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				MinItems:    1,
+				Description: "The droplets to apply the alert policy to",
+			},
+
+			"window": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"5m", "10m", "30m", "1h",
+				}, false),
+				MinItems: 1,
+			},
+		},
+	}
+}
+
+func resourceDigitalOceanMonitorAlertCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*CombinedConfig).godoClient()
+
+	alertCreateRequest := &godo.AlertPolicyCreateRequest{
+		Type:        d.Get("type").(string),
+		Description: d.Get("description").(string),
+		Tags:        expandTags(d.Get("tags").(*schema.Set).List()),
+		Compare:     d.Get("compare").(godo.AlertPolicyComp),
+		Window:      d.Get("window").(string),
+		Value:       d.Get("value").(float32),
+		Entities:    expandEntities(d.Get("entities").(*schema.Set).List()),
+	}
+
+	alertCreateRequest.Alerts = expandAlerts(d.Get("alerts").([]interface{}))
+
+	// alertPolicy, resp, err
+	alertPolicy, _, err := client.Monitoring.CreateAlertPolicy(ctx, alertCreateRequest)
+
+	if err != nil {
+		return diag.Errorf("Error creating Alert Policy: %s", err)
+	}
+
+	d.SetId(alertPolicy.UUID)
+	log.Printf("[INFO] Alert Policy created, ID: %s", d.Id())
+
+	return resourceDigitalOceanMonitorAlertRead(ctx, d, meta)
+}
+
+func expandAlerts(d *schema.ResourceData) *godo.Alerts {
+
+	alerts := &godo.Alerts{
+		Slack: expandSlack(d.Get("slack").([]interface{})),
+		Email: expandEmail(d.Get("email").([]interface{})),
+	}
```

Comment:
```suggestion
func expandAlerts(config []interface{}) godo.Alerts {
	alertConfig := config[0].(map[string]interface{})
	alerts := godo.Alerts{
		Slack: expandSlack(alertConfig["slack"].([]interface{})),
		Email: expandEmail(alertConfig["email"].([]interface{})),
```

Unfortunately, you can't just `d.Get("slack")` for these nested attributes.